### ISSUE TITLE
describeRegions not fetch all regions (missing ap-east-1 - Asia Pacific (Hong Kong))

### ISSUE THREAD

**Issue by shaikezam** [aws/aws-sdk-java#1995]
Hi,
Following the next code, I want to fetch the describe regions from **us-west-2**:

```
    private void fetchDescribeRegionsResult() {
    	AmazonEC2 ec2 = AmazonEC2Client.builder().withRegion(Regions.DEFAULT_REGION).build();

        DescribeRegionsResult regions_response = ec2.describeRegions();

        for(Region region : regions_response.getRegions()) {
            System.out.printf(
                "Found region %s " +
                "with endpoint %s",
                region.getRegionName(),
                region.getEndpoint());
        }
    }
```

This code will fetch me only 16 **but** the region **ap-east-1 - Asia Pacific (Hong Kong)** not among the 16.

I'm using with SDK version 1.11.545.
Is it known issue? or something wrong with the code?

Edit: issue happened also in 1.11.552

Thanks,

**Comment by debora-ito** [comment]
Hi @shaikezam, thank you for reaching out to us. The EC2 team is aware of the issue, and they are working on it. I'll update here if I get more information.

**Comment by shaikezam** [comment]
Hi,
Any update? or an estimation for fix?

**Comment by debora-ito** [comment]
Still no updates by EC2, unfortunately.

**Comment by korzakovYa91** [comment]
Hi,
any update on this issue?

**Comment by korzakovYa91** [comment]
Hi,
again, updates?

**Comment by shaikezam** [comment]
Hi there.
Any updates?

**Comment by debora-ito** [comment]
@shaikezam @korzakovYa91 

EC2 released a change on their DescribeRegions API. It will by default return information only about the Regions enabled for that account. They also added a new parameter `allRegions`, that indicates whether you want it to display all regions, including the ones disabled for your account. See the [API reference](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRegions.html) for more info.

The change is available in version `1.11.598` of the SDK.

I'll go ahead and close this. Feel free to reopen if you have further questions.
